File: Backend/app/Models/Vendor.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Vendor extends Model
{
    /** @use HasFactory<\Database\Factories\VendorFactory> */
    use HasFactory;

    protected $fillable = [
        'user_id',
        'name',
        'description',
        'total_rating',
        'contact_number',
        'address',
        'email',
    ];

    public function user()
    {
        return $this->belongsTo(User::class);
    }
    
    public function products()
    {
        return $this->hasMany(Product::class);
    }
}

File: Flutter/lib/data/models/vendor.dart
class Vendor {
  int? id;
  String? name;
  String? contactNumber;
  String? address;
  String? email;

  Vendor({this.id, this.name, this.contactNumber, this.address, this.email});

  factory Vendor.fromJson(Map<String, dynamic> json) {
    return Vendor(
      id: json['id'],
      name: json['name'],
      contactNumber: json['contactNumber'],
      address: json['address'],
      email: json['email'],
    );
  }
}